internal/services/oracle/oracle_gi_versions_data_source.go

```diff
@@ -7,23 +7,42 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"github.com/hashicorp/go-azure-helpers/lang/response"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2025-03-01/giversions"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
 )
 
 type GiVersionsDataSource struct{}
 
 type GiVersionsModel struct {
 	Versions []string `tfschema:"versions"`
 	Location string   `tfschema:"location"`
+	Shape    string   `tfschema:"shape"`
+	Zone     string   `tfschema:"zone"`
 }
 
 func (d GiVersionsDataSource) Arguments() map[string]*pluginsdk.Schema {
 	return map[string]*pluginsdk.Schema{
 		"location": commonschema.Location(),
+		"shape": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(giversions.SystemShapesExaDbXS),
+				string(giversions.SystemShapesExadataPointXNineM),
+				string(giversions.SystemShapesExadataPointXOneOneM),
+			}, false),
```

Comment:
We can use the `PossibleValuesFor...` helper here

```suggestion
			ValidateFunc: validation.StringInSlice(giversions.PossibleValuesForSystemShapes(), false),
```